apps/web/graphql/withApollo.tsx

```diff
@@ -1,12 +1,24 @@
 import React from 'react'
+import { useRouter } from 'next/router'
 import { NextPageContext } from 'next'
 import { ApolloProvider } from 'react-apollo'
+import { getLocaleFromPath } from '@island.is/web/i18n/withLocale'
 import initApollo from './client'
 
 export const withApollo = (Component) => {
   const NewComponent = ({ apolloState, pageProps }) => {
+    const Router = useRouter()
+
+    const { asPath } = Router
+
+    let clientLocale = null
+
+    if (asPath) {
+      clientLocale = getLocaleFromPath(asPath)
```

Comment:
so clientLocale could be null if `asPath` is falsy. Is that alright? should it be defaulted to Icelandic instead of being null?

Reply:
Right, I have made some changes here (there was no need to check for asPath).